beluga/include/beluga/motion/differential_drive_model.hpp

```diff
@@ -68,16 +69,23 @@ struct DifferentialDriveModelParam {
 
 /// Sampled odometry model for a differential drive.
 /**
+ * Supports 2D and (flattened) 3D state types.
  * This class satisfies \ref MotionModelPage.
  *
  * See Probabilistic Robotics \cite thrun2005probabilistic Chapter 5.4.2.
+ *
+ * \tparam NDim State dimension for the state representation. Only 2 and 3 are supported.
  */
+template <int NDim = 2>
```

Comment:
meta: I would prefer specializing with the state type, rather than the number of dimensions, for consistency with other 2D and 3D classes. And may also come in handy since it is directly related to the value type of the particle range.